android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/cookiebanners/GeckoCookieBannersStorage.kt

```diff
@@ -21,18 +23,32 @@ import org.mozilla.geckoview.StorageController
  */
 class GeckoCookieBannersStorage(
     runtime: GeckoRuntime,
+    context: Context,
 ) : CookieBannersStorage {
 
     private val geckoStorage: StorageController = runtime.storageController
     private val mainScope = CoroutineScope(Dispatchers.Main)
 
+    private val Context.dataStore by preferencesDataStore(
+        name = ReportSiteDomainsRepository.REPORT_SITE_DOMAINS_REPOSITORY_NAME,
+    )
+    private val reportSiteDomainsRepository = ReportSiteDomainsRepository(context.dataStore)
+
```

Comment:
Let's move this dependency into the constructor to be able to test `isSiteDomainReported` and `saveSiteDomain`. We just want to verify that everything we call `GeckoCookieBannersStorage.isSiteDomainReported` and `GeckoCookieBannersStorage.saveSiteDomain`  we are delegating to `ReportSiteDomainsRepository`